spark3-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestMergeIntoTable.java

```diff
@@ -0,0 +1,267 @@
+    createAndInitUnPartitionedTargetTable(targetName);
+    createAndInitSourceTable(sourceName);
+    append(sourceName, new Employee(1, "emp-id-1"), new Employee(2, "emp-id-2"), new Employee(3, "emp-id-3"));
+    String sqlText = "MERGE INTO " + targetName + " AS target \n" +
+                     "USING " + sourceName + " AS source \n" +
+                     "ON target.id = source.id \n" +
+                     "WHEN NOT MATCHED THEN INSERT * ";
+
+    sql(sqlText, "");
+    sql("SELECT * FROM %s ORDER BY id, dep", targetName);
+    assertEquals("Should have expected rows",
+            ImmutableList.of(row(1, "emp-id-1"), row(2, "emp-id-2"), row(3, "emp-id-3")),
+            sql("SELECT * FROM %s ORDER BY id ASC NULLS LAST", targetName));
+  }
+
+  @Test
+  public void testEmptyTargetInsertOnlyMatchingRows() throws NoSuchTableException {
+    createAndInitUnPartitionedTargetTable(targetName);
+    createAndInitSourceTable(sourceName);
+    append(sourceName, new Employee(1, "emp-id-1"), new Employee(2, "emp-id-2"), new Employee(3, "emp-id-3"));
+    String sqlText = "MERGE INTO " + targetName + " AS target \n" +
+                     "USING " + sourceName + " AS source \n" +
+                     "ON target.id = source.id \n" +
+                     "WHEN NOT MATCHED AND (source.id >= 2) THEN INSERT * ";
+
+    sql(sqlText, "");
+    List<Object[]> res = sql("SELECT * FROM %s ORDER BY id, dep", targetName);
+    assertEquals("Should have expected rows",
+            ImmutableList.of(row(2, "emp-id-2"), row(3, "emp-id-3")),
+            sql("SELECT * FROM %s ORDER BY id ASC NULLS LAST", targetName));
+  }
+
+  @Test
+  public void testOnlyUpdate() throws NoSuchTableException {
+    createAndInitUnPartitionedTargetTable(targetName);
+    createAndInitSourceTable(sourceName);
+    append(targetName, new Employee(1, "emp-id-one"), new Employee(6, "emp-id-6"));
+    append(sourceName, new Employee(2, "emp-id-2"), new Employee(1, "emp-id-1"), new Employee(6, "emp-id-6"));
+    String sqlText = "MERGE INTO " + targetName + " AS target \n" +
+            "USING " + sourceName + " AS source \n" +
+            "ON target.id = source.id \n" +
+            "WHEN MATCHED AND target.id = 1 THEN UPDATE SET * ";
+
+    sql(sqlText, "");
+    List<Object[]> res = sql("SELECT * FROM %s ORDER BY id, dep", targetName);
+    assertEquals("Should have expected rows",
+            ImmutableList.of(row(1, "emp-id-1"), row(6, "emp-id-6")),
+            sql("SELECT * FROM %s ORDER BY id ASC NULLS LAST", targetName));
+  }
+
+  @Test
+  public void testOnlyDelete() throws NoSuchTableException {
+    createAndInitUnPartitionedTargetTable(targetName);
+    createAndInitSourceTable(sourceName);
+    append(targetName, new Employee(1, "emp-id-one"), new Employee(6, "emp-id-6"));
+    append(sourceName, new Employee(2, "emp-id-2"), new Employee(1, "emp-id-1"), new Employee(6, "emp-id-6"));
+    String sqlText = "MERGE INTO " + targetName + " AS target \n" +
```

Comment:
Why did you choose to include the newlines?

Reply:
@rdblue i guess, its not required. I thought if we print out the SQL as part of some error or debugging, it will just format better. I will remove the newlines.

Reply:
I'm fine either way, it was just a surprise.